src/NuGet.Clients/NuGet.CommandLine/Common/NuGetTestMode.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Common
+{
+    // Note: we may delete this logic once we no longer depend on NuGet.Core's PackageServer type,
+    // as it is embedded in NuGet.Protocol.Core.Types.UserAgent.
+    internal static class NuGetTestMode
+    {
+        private const string _testModeEnvironmentVariableName = "NuGetTestModeEnabled";
```

Comment:
so you have confirmed that after functional tests finished, there is no NuGetTestModeEnabled variable set to the machine?

$env:NuGetTestModeEnabled = "True"